File: source/weha/wikianalysis/categorizer/PunctuationCorrection.java
package mo.umac.wikianalysis.categorizer;

import java.util.ArrayList;
import java.util.List;

import mo.umac.wikianalysis.diff.token.*;
import mo.umac.wikianalysis.lexer.MediawikiScannerConstants;
import mo.umac.wikianalysis.lexer.WikiToken;

public class PunctuationCorrection extends AbstractEditAction {

	{
		weight = 0.5;
	}
	
	public PunctuationCorrection() {
		this.be = null;
	}
	
	public PunctuationCorrection(BasicEdit b) {
		this.be = new BasicEdit[1];
		this.be[0] = b;
	}

	public boolean isAction(BasicEdit edit) {
		
		if ((edit instanceof Match) || (edit instanceof Movement))
		{
			return false;
		}
		else if (edit instanceof Replacement)
		{
			Replacement replEdit = (Replacement) edit;
			WikiToken[] insContent = replEdit.getInsertedContent();
			WikiToken[] delContent = replEdit.getDeletedContent();
			
			for (int i = 0; i < insContent.length; i++)
				if (insContent[i].kind != MediawikiScannerConstants.SYMBOL)
					return false;

			for (int i = 0; i < delContent.length; i++)
				if (delContent[i].kind != MediawikiScannerConstants.SYMBOL)
					return false;
		}
		else if (edit instanceof Insertion)
		{
			Insertion insEdit = (Insertion) edit;
			WikiToken[] insContent = insEdit.getContent();
			
			for (int i = 0; i < insContent.length; i++)
				if (insContent[i].kind != MediawikiScannerConstants.SYMBOL)
					return false;
		}
		else if (edit instanceof Deletion)
		{
			Deletion delEdit = (Deletion) edit;
			WikiToken[] delContent = delEdit.getContent();

			for (int i = 0; i < delContent.length; i++)
				if (delContent[i].kind != MediawikiScannerConstants.SYMBOL)
					return false;
		}
		
		return true;
	}
	
	@Override
	public List<AbstractEditAction> classify(List<BasicEdit> editList) {
		ArrayList<AbstractEditAction> ret = new ArrayList<AbstractEditAction>();
		ArrayList<BasicEdit> newEditList = new ArrayList<BasicEdit>();
		
		for (int i = 0; i < editList.size(); i++)
		{
			BasicEdit b = editList.get(i);
			if (this.isAction(b))
				ret.add(new PunctuationCorrection(b));
			else {
				newEditList.add(b);
			}
		}
		
		editList.clear();
		editList.addAll(newEditList);
		
		return ret;
	}

	@Override
	public int lengthCount() {
		return 1;
	}

}

File: weha/source/weha/categorizer/PunctuationCorrection.php
<?php
require_once('AbstractEditAction.php');

class PunctuationCorrection extends AbstractEditAction {

	public function __construct($b = null) {
		$this->be = array();
		$this->be[] = $b;
	}

	public function isAction($edit) {
		
		if (($edit instanceof Match) || ($edit instanceof Movement))
		{
			return false;
		}
		elseif ($edit instanceof Replacement)
		{
			$replEdit = $edit;
			$insContent = $replEdit->getInsertedContent();
			$delContent = $replEdit->getDeletedContent();
			
			for ($i = 0; $i < count($insContent); $i++)
				if ($insContent[$i]->kind != 
					WikiLexerConstants::SYMBOL)
					return false;

			for ($i = 0; $i < count($delContent); $i++)
				if ($delContent[$i]->kind != 
					WikiLexerConstants::SYMBOL)
					return false;
		}
		elseif ($edit instanceof Insertion)
		{
			$insEdit = $edit;
			$insContent = $insEdit->getContent();
			
			for ($i = 0; $i < count($insContent); $i++)
				if ($insContent[$i]->kind != 
					WikiLexerConstants::SYMBOL)
					return false;
		}
		elseif ($edit instanceof Deletion)
		{
			$delEdit = $edit;
			$delContent = $delEdit->getContent();

			for ($i = 0; $i < count($delContent); $i++)
				if ($delContent[$i]->kind != 
					WikiLexerConstants::SYMBOL)
					return false;
		}
		
		return true;
	}
	
	public function classify(&$editList) {
		$ret = array();
		$newEditList = array();
		
		for ($i = 0; $i < count($editList); $i++)
		{
			$b = $editList[$i];
			if ($this->isAction($b))
				$ret[] = new PunctuationCorrection($b);
			else {
				$newEditList[] = $b;
			}
		}
		
		$editList = $newEditList;
		
		return $ret;
	}

}
